File: arrays/2sum.py
import sys

def main():
    lst = sys.argv[0]
    target = sys.argv[1]
    def twoSum(lst, target):
        hm = {}
        for i, v in enumerate(lst):
            diff = target - lst[i]
            print("diff", diff, "list[i]", lst[i])
            if diff in hm:
                return [hm[v], i]
            hm[i] = diff

File: 2sum.rb
def two_sum nums, target
  hm = Hash.new
  nums.each.with_index do |num, index|
    diff = hm[target - num]
    return [diff, index] if diff != nil
  hm[num] = index
  end
end

puts two_sum [2,3,4,5,6], 11
